EpikCord/client/buckets.py

```diff
@@ -0,0 +1,181 @@
+from __future__ import annotations
+
+from ..utils import clear_none_values
+
+import asyncio
+from logging import getLogger
+from typing import Union, Optional, Dict
+
+import aiohttp
+
+logger = getLogger("EpikCord.http")
+
+class MockBucket:
+    async def wait(self):
+        ...
+
+    def clear(self):
+        ...
+
+    def set(self):
+        ...
+
+    def __eq__(self, other: MockBucket):
+        return isinstance(other, MockBucket)
+
+class Bucket:
+    def __init__(self, *, bucket_hash: str):
+        """
+        Parameters
+        ----------
+        bucket_hash: str
+            The bucket hash of the bucket provided by Discord.
+
+        Attributes
+        ----------
+        bucket_hash: str
+            The bucket hash of the bucket provided by Discord.
+        event: asyncio.Event
+            The event that is used to wait for the bucket to be set.
+        """
+        self.hash: str = bucket_hash
+        self.event: asyncio.Event = asyncio.Event()
+        self.event.set()
+
+    async def wait(self):
+        """
+        Waits for the bucket to be set.
+        """
+        logger.info(f"Waiting for bucket {self.hash} to be set.")
+        await self.event.wait()
+        logger.info(f"Done waiting for bucket {self.hash}.")
+
+    def set(self):
+        """
+        Sets the bucket.
+        """
+        logger.info(f"Setting bucket {self.hash}.")
+        self.event.set()
+
+    def clear(self):
+        """
+        Clears the bucket.
+        """
+        logger.info(f"Clearing bucket {self.hash}.")
+        self.event.clear()
+
+    def __eq__(self, other: Bucket):
+        if isinstance(other, Bucket):
+            return self.hash == other.hash
+        return False
```

Comment:
other should always be `Any` as it could be compared to anything
```suggestion
    def __eq__(self, other: Any) -> bool:
        if isinstance(other, Bucket):
            return self.hash == other.hash
        return False
```